tests/foreman/ui/test_bookmarks.py

```diff
@@ -30,53 +28,44 @@
 from robottelo.utils.issue_handlers import is_open
 
 
-@pytest.fixture(scope='module')
-def ui_entities(module_org, module_location):
+@pytest.fixture(
+    scope='module', params=BOOKMARK_ENTITIES, ids=(i['name'] for i in BOOKMARK_ENTITIES)
+)
+def ui_entity(module_org, module_location, request):
     """Collects the list of all applicable UI entities for testing and does all
     required preconditions.
     """
-    ui_entities = []
-    for entity in BOOKMARK_ENTITIES:
+    entity = request.param
+    # Some pages require at least 1 existing entity for search bar to
+    # appear. Creating 1 entity for such pages
+    entity_name, entity_setup = entity['name'], entity.get('setup')
+    if entity_setup:
         # Skip the entities, which can't be tested ATM (not implemented in
         # airgun or have open BZs)
         skip = entity.get('skip_for_ui')
         if isinstance(skip, (tuple, list)):
-            skip = any([is_open(issue) for issue in skip])
-        if skip is True:
-            continue
-        ui_entities.append(entity)
-        # Some pages require at least 1 existing entity for search bar to
-        # appear. Creating 1 entity for such pages
-        entity_name, entity_setup = entity['name'], entity.get('setup')
-        if entity_setup:
-            # entities with 1 organization and location
-            if entity_name in ('Host',):
-                entity_setup(organization=module_org, location=module_location).create()
-            # entities with no organizations and locations
-            elif entity_name in (
-                'ComputeProfile',
-                'ConfigGroup',
-                'GlobalParameter',
-                'HardwareModel',
-                'PuppetClass',
-                'UserGroup',
-            ):
-                entity_setup().create()
-            # entities with multiple organizations and locations
-            else:
-                entity_setup(organization=[module_org], location=[module_location]).create()
-    return ui_entities
-
-
-@pytest.fixture()
-def random_entity(ui_entities):
-    """Returns one random entity from list of available UI entities"""
-    return ui_entities[random.randint(0, len(ui_entities) - 1)]
+            issues = {is_open(issue) for issue in skip}
+            pytest.skip(f'There is/are an open issue(s) with this entity: {issues}')
+        # entities with 1 organization and location
+        if entity_name in ('Host',):
+            entity_setup(organization=module_org, location=module_location).create()
+        # entities with no organizations and locations
+        elif entity_name in (
```

Comment:
Might be nice to put this in the datastructure for the entities, so the information about module/org context isn't built into the logic here.

Reply:
This is true. I would like to modify the data-structure. What is more important for us now is to make all the tests passing, isn't it? Sorry but I have to postpone this request bit towards the end of the universe with hope there will be time and me wise enough to do it lean and neat way. (:

Reply:
Definitely fine to not include that change in this PR!